tests/integration/ha_tests/test_rollback_to_master_label.py

```diff
@@ -0,0 +1,174 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import asyncio
+import logging
+import operator
+import shutil
+from pathlib import Path
+
+import pytest
+from pytest_operator.plugin import OpsTest
+from tenacity import Retrying, stop_after_attempt, wait_fixed
+
+from .. import markers
+from ..helpers import (
+    APPLICATION_NAME,
+    DATABASE_APP_NAME,
+    get_leader_unit,
+    get_primary,
+    get_unit_by_index,
+)
+from .helpers import (
+    are_writes_increasing,
+    check_writes,
+    get_instances_roles,
+    inject_dependency_fault,
+    start_continuous_writes,
+)
+
+logger = logging.getLogger(__name__)
+
+TIMEOUT = 600
+
+
+@pytest.mark.group(1)
+@markers.amd64_only  # TODO: remove after arm64 stable release
+@pytest.mark.abort_on_fail
+async def test_deploy_stable(ops_test: OpsTest) -> None:
+    """Simple test to ensure that the PostgreSQL and application charms get deployed."""
+    await asyncio.gather(
+        ops_test.model.deploy(
+            DATABASE_APP_NAME,
+            num_units=3,
+            channel="14/stable",
```

Comment:
Maybe we should use a specific revision here? New releases would use the new label by default.